### PR TITLE
dbeaver#add test case for PostgreEscapeStringRule.java

### PR DESCRIPTION
**Pull request by TanishaAAgarwal4**
Added test case for the evaluate function in the PostgreEscapeStringRule file in the PostgreEscapeStringRuleTest

### CODE REVIEW

**Comment by E1izabeth** [comment]
Please, use 4 spaces instead of tabs.

**Comment by LonwoLonwo** [comment]
Hello @TanishaAAgarwal4 

First of all - your file doesn't have an extension. It will not work without it.
Second - you didn't add a copyright on the top (you can check it in any other test file).
Third - please fix the formatting in the file. Now your code is in strange positions.
Fourth - we avoid using `System.out.println(e)`. We use logging instead. Examples you can also find in other test files.

**Comment by TanishaAAgarwal4** [comment]
> Hello @TanishaAAgarwal4
> 
> First of all - your file doesn't have an extension. It will not work without it. Second - you didn't add a copyright on the top (you can check it in any other test file). Third - please fix the formatting in the file. Now your code is in strange positions. Fourth - we avoid using `System.out.println(e)`. We use logging instead. Examples you can also find in other test files.

What do you mean by file extension?

**Comment by E1izabeth** [comment]
Eacch java file should have .java extension.

File: test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+         // Assert that the result matches the expected token
+         assertEquals(SQLTokenType.T_STRING, result);
+
+         // Verify that the expected methods were called on the scanner
+         verify(scannerMock).getColumn();
+         verify(scannerMock, times(4)).read();
+         verify(scannerMock, times(2)).unread();
+         } catch (Exception e) {
+         e.printStackTrace();
```
Comment (LonwoLonwo) [inline test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest]: Please use logging instead. Examples you can find in other files.

Comment (LonwoLonwo) [reply test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest]: And please fix the incorrect formatting here.

**Comment by E1izabeth** [comment]
Please look at SQLScriptParserTest:parseOracleQStringRule. You can find example of how to write test for something like this. Currently it looks like a test just to write and doesn't bring some meaningful value.

Besides that, this can't be compiled. See the log part below.
```
Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:3.0.1:compile (default-compile) on project org.jkiss.dbeaver.ext.postgresql.test: Compilation failure: Compilation failure: 
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[24] 
[ERROR] 	import org.slf4j.Logger;
[ERROR] 	       ^^^^^^^^^^^^^^^^
[ERROR] The import org.slf4j.Logger cannot be resolved
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[25] 
[ERROR] 	import org.slf4j.LoggerFactory;
[ERROR] 	       ^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The import org.slf4j.LoggerFactory cannot be resolved
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[33] 
[ERROR] 	private static final Logger log = LoggerFactory.getLogger(PostgreEscapeStringRuleTest.class);
[ERROR] 	                     ^^^^^^
[ERROR] Logger cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[33] 
[ERROR] 	private static final Logger log = LoggerFactory.getLogger(PostgreEscapeStringRuleTest.class);
[ERROR] 	                                  ^^^^^^^^^^^^^
[ERROR] LoggerFactory cannot be resolved
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[36] 
[ERROR] 	TPCharacterScanner scannerMock;
[ERROR] 	^^^^^^^^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[46] 
[ERROR] 	when(scannerMock.getColumn()).thenReturn(0);
[ERROR] 	     ^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[47] 
[ERROR] 	when(scannerMock.read()).thenReturn(TPCharacterScanner.EOF);
[ERROR] 	     ^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[47] 
[ERROR] 	when(scannerMock.read()).thenReturn(TPCharacterScanner.EOF);
[ERROR] 	                                    ^^^^^^^^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a variable
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[50] 
[ERROR] 	TPToken result = obj.evaluate(scannerMock, resume);
[ERROR] 	^^^^^^^
[ERROR] TPToken cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[50] 
[ERROR] 	TPToken result = obj.evaluate(scannerMock, resume);
[ERROR] 	                              ^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[56] 
[ERROR] 	verify(scannerMock).getColumn();
[ERROR] 	       ^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[57] 
[ERROR] 	verify(scannerMock, times(4)).read();
[ERROR] 	       ^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[58] 
[ERROR] 	verify(scannerMock, times(2)).unread();
[ERROR] 	       ^^^^^^^^^^^
[ERROR] TPCharacterScanner cannot be resolved to a type
[ERROR] /var/jenkins_home/jobs/DBeaverDesktop/jobs/dbeaver-ce-github-checks/branches/PR-20051/workspace/dbeaver/test/org.jkiss.dbeaver.ext.postgresql.test/src/sql/PostgreEscapeStringRuleTest.java:[60] 
[ERROR] 	log.error("Error" + e);
[ERROR] 	^^^
[ERROR] Logger cannot be resolved to a type
[ERROR] 14 problems (14 errors)
```

**Comment by LonwoLonwo** [comment]
Ok, could you please describe - what do you test? What is the main goal of this code?

**Comment by E1izabeth** [comment]
This PR is stale, so I'm closing it. Please, let me know if you'll continue working on this - we can reopen it.

**Pull request closed without merging**